lib/record_store/provider/ns1.rb

```diff
@@ -5,6 +5,39 @@ module RecordStore
   class Provider::NS1 < Provider
     class Error < StandardError; end
 
+    class ApiAnswer
+      def self.from_full_api_answer(type:, record_id:, answer:)
+        ApiAnswer.new(type: type, record_id: record_id, rrdata: answer["answer"])
+      end
+
+      def self.from_short_api_answer(type:, record_id:, answer:)
+        rrdata_fields = case type
+        when 'SPF', 'TXT'
+          [answer]
+        else
+          answer.split
+        end
+
+        ApiAnswer.new(type: type, record_id: record_id, rrdata: rrdata_fields)
+      end
+
+      attr_accessor :type, :record_id, :rrdata
+
+      def initialize(type:, record_id:, rrdata:)
+        @type = type
+        @record_id = record_id
+        @rrdata = rrdata
+      end
+
+      def rrdata_string
+        rrdata.join(' ')
+      end
+
+      def id
+        [record_id, type, rrdata.join(':')].join(':')
```

Comment:
It can be anything comparable, but so it's super clear what the digest algorithm is, I opted for a simple string that clearly incorporates these fields, but you're right: An Array would work, and they can be compared in Ruby.

I'll double check if anything assumes the type of an `id` and change to an Array / remove the stringification.

Reply:
Disregarding perf, ur right that an object could work here.  These ids are just used in-memory, so they can be anything.  I'll just use an Array for simplicity.